e2e/psr-4/rector.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+use Rector\Set\ValueObject\SetList;
+use Rector\PSR4\Rector\FileWithoutNamespace\NormalizeNamespaceByPSR4ComposerAutoloadRector;
+use Rector\PostRector\Rector\ClassRenamingPostRector;
+
+return static function (Rector\Config\RectorConfig $rectorConfig): void {
+    $rectorConfig->paths([
+        __DIR__ . '/src',
+    ]);
+
+    $rectorConfig->rule(NormalizeNamespaceByPSR4ComposerAutoloadRector::class);
+    $rectorConfig->rule(ClassRenamingPostRector::class);
```

Comment:
Not sure if this is needed but it does not work either way – it does not look like the `ClassRenamingPostRector` does anything to fully-qualified name references.

Reply:
Apparently, it is not necessary. And it is skipping the processing here:

https://github.com/rectorphp/rector-src/blob/02cec7ebc43469d599cc6a66945e239267626ed6/rules/Renaming/NodeManipulator/ClassRenamer.php#L166

Reply:
Apparently, the `$parentNode` is `null`. Possibly `FullyQualifyStmtsAnalyzer` is removing them?

Reply:
`FullyQualifyStmtsAnalyzer->process()` produce create new `FullyQualified` instance, the attribute may need to be mirrored:

```php
return new FullyQualified($name, $node->getAttributes());
```

Reply:
Even then, it still does not seem to work properly – only fixes references in one of the files (weird, they are structurally identical). Will need to do more digging.

Reply:
I am using `RenamedClassesDataCollector` now and it updates the references in one of the files correctly but not the other.